### PR TITLE
backport(2.4): add permission related to monitoring and missing label on namespaces

### PR DESCRIPTION
**Pull request by zdtsw**
## Description

as part of https://github.com/red-hat-data-services/rhods-operator/pull/98
logic part will be added into https://github.com/opendatahub-io/opendatahub-operator/pull/618

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by zdtsw** [comment]
do not merge till https://github.com/opendatahub-io/opendatahub-operator/pull/618 is in first

File: controllers/dscinitialization/utils.go [hunk 1]
```diff
@@ -66,9 +66,11 @@ func (r *DSCInitializationReconciler) createOdhNamespace(ctx context.Context, ds
 			r.Log.Error(err, "Unable to fetch namespace", "name", name)
 			return err
 		}
-	} else if (dscInit.Spec.Monitoring.ManagementState == operatorv1.Managed) && dscInit.Spec.Monitoring.Namespace == name {
+	} else if dscInit.Spec.Monitoring.ManagementState == operatorv1.Managed {
+		r.Log.Info("Patching application namespace for Managed cluster", "name", name)
+		lablePatch := `{"metadata":{"labels":{"openshift.io/cluster-monitoring":"true","pod-security.kubernetes.io/enforce":"baseline","opendatahub.io/generated-namespace": "true"}}}` //nolint
```
Comment (ykaliuta) [inline controllers/dscinitialization/utils.go:71]: Did you mean `labelPatch`?

File: controllers/dscinitialization/utils.go [hunk 2]
```diff
@@ -100,6 +102,14 @@ func (r *DSCInitializationReconciler) createOdhNamespace(ctx context.Context, ds
 				r.Log.Error(err, "Unable to fetch monitoring namespace", "name", monitoringName)
 				return err
 			}
+		} else { // force to patch monitoring namespace with label for cluster-monitoring
+			r.Log.Info("Patching monitoring namespace for Managed cluster", "name", monitoringName)
```
Comment (ykaliuta) [inline controllers/dscinitialization/utils.go:106]: I guess such a code duplication fits the general style, doesn't it?

File: controllers/dscinitialization/utils.go (hunk 1 above)
Comment (zdtsw) [reply controllers/dscinitialization/utils.go:71]: correct, this is a typo for the variable name

File: controllers/dscinitialization/utils.go (hunk 2 above)
Comment (zdtsw) [reply controllers/dscinitialization/utils.go:106]: it is mostly backport for the same code from downstream to odh.
and to refine the code, we can do have another PR to follow it up.
as for duplication, for ODH itself these labels may not be needed at all so the logic  is only needed to adopt for downstream.

**Review by ykaliuta — APPROVED** [review]
Thanks!

**Pull request merged**